common/djangoapps/entitlements/api/v1/tests/test_views.py

```diff
@@ -195,3 +198,168 @@ def test_revoke_unenroll_entitlement(self):
         course_entitlement.refresh_from_db()
         assert course_entitlement.expired_at is not None
         assert course_entitlement.enrollment_course_run is None
+
+
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+class EntitlementEnrollmentViewSetTest(ModuleStoreTestCase):
+    ENTITLEMENTS_ENROLLMENT_NAMESPACE = 'entitlements_api:v1:enrollments'
+
+    def setUp(self):
+        super(EntitlementEnrollmentViewSetTest, self).setUp()
+        self.user = UserFactory()
+        self.client.login(username=self.user.username, password=TEST_PASSWORD)
+        self.course = CourseFactory.create(org='edX', number='DemoX', display_name='Demo_Course')
+        self.course2 = CourseFactory.create(org='edX', number='DemoX2', display_name='Demo_Course 2')
+        self.patcher = patch("openedx.core.djangoapps.catalog.utils.get_course_runs_for_course")
+        self.mock_get_course_runs = self.patcher.start()
+        self.mock_get_course_runs.return_value = [
+            {'key': str(self.course.id)},
+            {'key': str(self.course2.id)}
+        ]
+
+    def tearDown(self):
+        self.patcher.stop()
+        super(EntitlementEnrollmentViewSetTest, self).tearDown()
+
+    def test_user_can_enroll(self):
+        course_entitlement = CourseEntitlementFactory(
+            user=self.user
+        )
+        url = reverse(
+            self.ENTITLEMENTS_ENROLLMENT_NAMESPACE,
+            args=[str(course_entitlement.uuid)]
+        )
+        assert course_entitlement.enrollment_course_run is None
+
+        data = {
+            'course_run_id': str(self.course.id)
+        }
+        response = self.client.post(
+            url,
+            data=json.dumps(data),
+            content_type='application/json',
+        )
+        course_entitlement.refresh_from_db()  # pylint: disable=no-member
```

Comment:
I don't think you need to disable the warning here if you swap `CourseEntitlementFactory()` to use `CourseEntitlementFactory.create()`